shared/util/tf_parser.go

```diff
@@ -0,0 +1,74 @@
+package util
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/pkg/errors"
+)
+
+func ParseServices(dir string) (map[string]bool, error) {
+	var services map[string]bool = map[string]bool{}
+	schema := &hcl.BodySchema{
+		Blocks: []hcl.BlockHeaderSchema{
+			{
+				Type:       "module",
+				LabelNames: []string{"name"},
+			},
+		},
+	}
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			if info.Name() == ".terraform" || info.Name() == ".git" {
+				return filepath.SkipDir
+			}
+		}
+		if !info.IsDir() && filepath.Ext(path) == ".tf" {
+			b, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			f, diags := hclsyntax.ParseConfig(b, path, hcl.Pos{Line: 1, Column: 1})
+			if diags.HasErrors() {
+				return errors.Wrapf(diags.Errs()[0], "failed to parse %s", path)
+			}
+
+			content, _, contentDiags := f.Body.PartialContent(schema)
+			if contentDiags.HasErrors() {
+				return errors.New("Terraform code has errors")
+			}
+			for _, block := range content.Blocks {
```

Comment:
can you write this in a way that doesn't do the nesting like this? Essentially have the business logic in the same level of indenting, and if its an error continue the loop. Its harder to read like this

Reply:
aka. 

~~~golang
if block.Type != "module" {
  continue
}

if sourceAttr, ok := attrs["source"]; !ok {
 continue
}

// etc
//
// main business logic
~~~